tests/ProxyManagerTest/Factory/AbstractBaseFactoryTest.php

```diff
@@ -169,4 +170,41 @@ public function testGeneratesClass() : void
             $generateProxy->invoke($this->factory, stdClass::class, ['some' => 'proxy', 'options' => 'here'])
         );
     }
+
+    /**
+     * @group 469
+     */
+    public function testGeneratesClassBasedOnGivenProxyOptions() : void
+    {
+        $generateProxy = new ReflectionMethod($this->factory, 'generateProxy');
+
+        $generateProxy->setAccessible(true);
+        $generatedClass = UniqueIdentifierGenerator::getIdentifier('fooBar');
+        $proxyOptions   = [uniqid('key', true) => uniqid('value', true)];
+
+        $this
+            ->classNameInflector
+            ->expects(self::any())
+            ->method('getProxyClassName')
+            ->with('stdClass', self::callback(static function (array $parameters) use ($proxyOptions) : bool {
+                self::assertSame($proxyOptions, $parameters['proxyOptions']);
+
+                return true;
+            }))
+            ->will(self::returnValue($generatedClass));
+
+        $this
+            ->proxyAutoloader
+            ->method('__invoke')
+            ->will(self::returnCallback(function (string $className) : bool {
+                eval('class ' . $className . ' {}');
```

Comment:
Can't this be skipped?

Reply:
Cannot: a class must be created as part of the test here, or else it will fail to instantiate the proxy